### PR TITLE
#4 Feature/session per project

### PR DESCRIPTION
**Pull request by dvanderleij**
See #21

### CODE REVIEW

File: src/main/java/refactoringml/App.java [hunk 1]
```diff
@@ -251,15 +253,13 @@ private void processCommit(RevCommit currentCommit, GitHistoryRefactoringMiner m
 			Set<ImmutablePair<String, String>> jGitRenames = getJGitRenames(entries);
 			processMetrics.collectMetrics(currentCommit, superCommitMetaData, allRefactoringCommits, entries, refactoringRenames, jGitRenames);
 			long startTimeTransaction = System.currentTimeMillis();
-			db.commit();
+			t.commit();
 			log.debug("Committing the transaction for commit " + commitHash + " took " + (System.currentTimeMillis() - startTimeTransaction) + " milliseconds.");
 		} catch (Exception e) {
 			exceptionsCount++;
 			log.error("Unhandled exception when collecting commit data for commit: " + commitHash + createErrorState(commitHash, project), e);
 			db.rollback(createErrorState(commitHash, project));
```
Comment (jan-gerling) [inline src/main/java/refactoringml/App.java]: Is the rollback, still working as intended?

File: src/main/java/refactoringml/App.java [hunk 2]
```diff
@@ -210,9 +213,8 @@ private Project initProject(Git git) throws GitAPIException, IOException {
 	private void processCommit(RevCommit currentCommit, GitHistoryRefactoringMiner miner, RefactoringHandler handler, RefactoringAnalyzer refactoringAnalyzer, ProcessMetricsCollector processMetrics){
 		long startCommitTime = System.currentTimeMillis();
 		String commitHash = currentCommit.getId().getName();
+		Transaction t = db.beginTransaction();
```
Comment (jan-gerling) [inline src/main/java/refactoringml/App.java:223]: What happens if we cannot create a new transaction?

**Review by jan-gerling — COMMENTED** [review]
Looks very promising, but I have a few questions, see comments.

Also, can you please verify the stability of the changes to handling hibernate and the database with `project_list_cleaned_noDup.csv`, because `projects-final.csv` contains lots of non-java projects and is thereby, not really a good stress test. I will remove the file from the repo.

File: src/main/java/refactoringml/App.java (hunk 1 above)
Comment (dvanderleij) [reply src/main/java/refactoringml/App.java]: Yes i tested it by debugging, maybe an idea for the future to write a test for it though

File: src/main/java/refactoringml/App.java (hunk 2 above)
Comment (dvanderleij) [reply src/main/java/refactoringml/App.java:223]: I have added a try catch in the method above so it doesn't stop the whole project when one commit fails.

**Comment by dvanderleij** [comment]
> Looks very promising, but I have a few questions, see comments.
> 
> Also, can you please verify the stability of the changes to handling hibernate and the database with `project_list_cleaned_noDup.csv`, because `projects-final.csv` contains lots of non-java projects and is thereby, not really a good stress test. I will remove the file from the repo.

I have tested with that list. no issues in the first 200 or so projects

**Pull request merged**